MdePkg/Library/BaseRngLib/Riscv/Rng.c

```diff
@@ -0,0 +1,279 @@
+/** @file
+   Random number generator service that uses the SEED instruction
+   to provide pseudorandom numbers.
+
+   Copyright (c) 2024, Rivos, Inc.
+
+   SPDX-License-Identifier: BSD-2-Clause-Patent
+ **/
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/RngLib.h>
+
+#include "BaseRngLibInternals.h"
+#define RISCV_CPU_FEATURE_ZKR_BITMASK  0x8
+
+#define SEED_RETRY_LOOPS  100
+
+#define SEED_OPST_MASK     0xc0000000
+#define SEED_OPST_BIST     0x00000000
+#define SEED_OPST_WAIT     0x40000000
+#define SEED_OPST_ES16     0x80000000
+#define SEED_OPST_DEAD     0xc0000000
+#define SEED_ENTROPY_MASK  0xffff
+
+// 64-bit Mersenne Twister implementation

```

Comment:
Any reason to use this PRNG? Comment would help.

Reply:
Added brief.